packages/jest/src/matchers.tsx

```diff
@@ -15,15 +22,66 @@ const getMountedProperties = () =>
     )
     .join(' ');
 
-const containsClassNames = (classNames: string[], css: string) =>
-  classNames.reduce((accum, className) => (css.includes(`.${className}`) ? true : accum), false);
+const getRules = (ast: CSS.Stylesheet, filter: MatchFilter, className: string) => {
+  const { media, state } = filter;
+
+  // rules are present directly inside ast.stylesheet.rules
+  // but if a media query is present it is nested inside ast.stylesheet.media.rules
+  // this inner function returns the relevant rules
+  const getAllRules = () => {
+    if (media) {
+      const mediaRules = ast.stylesheet?.rules.filter(r => {
+        if ('media' in r) {
+          return r.media === media;
+        }
+        return;
+      });
+
+      return mediaRules?.reduce<CSS.Rule[]>((acc, m) => {
+        if ('rules' in m && m.rules) {
+          acc = [...acc, ...m.rules];
+        }
+        return acc;
+      }, []);
+    }
+    return ast.stylesheet?.rules.filter(r => (r.type = 'rule')); // omit media objects
+  };
+
+  const allRules = getAllRules();
+  const klass = state ? `.${className}:${state}` : `.${className}`;
```

Comment:
will we be able to assert or more complex selectors?

```
css={`
  > :first-child {
    color: red;
  }
`}

expect(element).toHaveCompiledCss('color', 'red', { state: '> :first-child' });
```

Does that work? If not should it?

Reply:
Wow, let me try this as well